pkg/operation/botanist/etcdcopybackupstask.go

```diff
@@ -0,0 +1,77 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package botanist
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gardener/gardener/extensions/pkg/controller/backupentry"
+	"github.com/gardener/gardener/extensions/pkg/controller/backupentry/genericactuator"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/etcdcopybackupstask"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// NewEtcdCopyBackupsTask is a function exposed for testing.
+var NewEtcdCopyBackupsTask = etcdcopybackupstask.New
+
+// DefaultEtcdCopyBackupsTask creates the default deployer for the EtcdCopyBackupsTask resource.
+func (b *Botanist) DefaultEtcdCopyBackupsTask() etcdcopybackupstask.Interface {
+	return NewEtcdCopyBackupsTask(
+		b.Logger,
+		b.K8sSeedClient.Client(),
+		&etcdcopybackupstask.Values{
+			Name:      b.Shoot.GetInfo().Name,
+			Namespace: b.Shoot.SeedNamespace,
+		},
+		etcdcopybackupstask.DefaultInterval,
+		etcdcopybackupstask.DefaultTimeout,
+	)
+}
+
+// DeployEtcdCopyBackupsTask sets the target and destination object stores of the EtcdCopyBackupsTask resource and deploys it.
+func (b *Botanist) DeployEtcdCopyBackupsTask(ctx context.Context) error {
+	sourceSecrtName := fmt.Sprintf("%s-%s", backupentry.SourcePrefix, genericactuator.BackupSecretName)
+	sourceSecret := &corev1.Secret{}
+	if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, sourceSecrtName), sourceSecret); err != nil {
+		return err
+	}
+	secret := &corev1.Secret{}
+	if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, genericactuator.BackupSecretName), secret); err != nil {
+		return err
+	}
+
+	provider := druidv1alpha1.StorageProvider(b.Seed.GetInfo().Spec.Backup.Provider)
+	sourceContainer := string(sourceSecret.Data[genericactuator.DataKeyBackupBucketName])
+	container := string(secret.Data[genericactuator.DataKeyBackupBucketName])
+
+	b.Shoot.Components.ControlPlane.EtcdCopyBackupsTask.SetSourceStore(druidv1alpha1.StoreSpec{
+		Provider:  &provider,
+		SecretRef: &corev1.SecretReference{Name: sourceSecret.Name},
+		Prefix:    fmt.Sprintf("%s/etcd-%s", b.Shoot.BackupEntryName, "main"),
+		Container: &sourceContainer,
+	})
+	b.Shoot.Components.ControlPlane.EtcdCopyBackupsTask.SetTargetStore(druidv1alpha1.StoreSpec{
+		Provider:  &provider,
+		SecretRef: &corev1.SecretReference{Name: secret.Name},
+		Prefix:    fmt.Sprintf("%s/etcd-%s", b.Shoot.BackupEntryName, "main"),
+		Container: &container,
+	})
+
+	return b.Shoot.Components.ControlPlane.EtcdCopyBackupsTask.Deploy(ctx)
```

Comment:
I would move the destroy / wait logic to here so it would look something like:
```suggestion
	if err := b.Shoot.Components.ControlPlane.EtcdCopyBackupsTask.Destroy(ctx); err != nil {
		return err
	}
	if err := b.Shoot.Components.ControlPlane.EtcdCopyBackupsTask.WaitCleanup(ctx); err != nil {
		return err
	}
	return b.Shoot.Components.ControlPlane.EtcdCopyBackupsTask.Deploy(ctx)
```